### PR TITLE
Update DG for Ui and enable assertions

### LINKED ISSUE #104: Update Ui Documentation
Update the documentation for the Ui and include explanation of the implementation for the GUI view switching functionality.

### PR DESCRIPTION
**Pull request by TanLeYang**
Closes #104 
Update the DG with implementation details and diagrams for the Ui, specifically how the Ui switches between the summarised view and  the detailed view.

Also enabled assertions for the project as per the requirements in this week's project dashboard.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -64,9 +64,10 @@ The sections below give more details of each component.
...
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `BookListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
 
-The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
+The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`].(https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
```
Comment (pennhanlee) [inline docs/DeveloperGuide.md]: Can you change the link of the MainWindow.fxml please?

**Review by pennhanlee — CHANGES_REQUESTED** [review]
Besides the link that needs to be updated, LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (TanLeYang) [reply docs/DeveloperGuide.md]: Thanks for spotting the mistake! I've made the necessary changes

**Review by angrybunny123 — APPROVED** [review]
LGTM!

**Review by mgiang2015 — APPROVED** [review]
LGTM

**Review by pennhanlee — COMMENTED** [review]
LGTM!

**Review by pennhanlee — APPROVED** [review]

**Pull request merged**